### DISCUSSION TITLE
send additional data in headers for UploadFile

### DISCUSSION THREAD

**Discussion by TamarLevy** [fastapi/fastapi#5478] [Questions]
Im trying to send an additional parameter in  headers in post request:
`          headers = {
                'Accept': 'application/json',
                'md5_file_integrity': src_file_md5
            }
            res = self.client.post(command_url, files=files, headers=headers)` 

to :

`@app.post("/upload_file")
async def  upload_file(file: UploadFile):
 file_buffer = file.file.read()
 if 'md5_file_integrity' in file.headers.keys():
        md5_file_integrity = file.headers['md5_file_integrity']
        local_md5_file_integrity = calculate_file_md5(file_buffer)
        if local_md5_file_integrity != md5_file_integrity:
            return Response('file integrity is incorrect', status_code=status.HTTP_507_INSUFFICIENT_STORAGE)

`
when upload_file is called there is no 'md5_file_integrity' field in headers.
does anyone know if its even possible to send such information in headers, or maybe its not a proper place (headers) to send such information (md5 integrity)?

**Comment by tiangolo** [comment]
Please ask in an issue following the template and providing a self-contained example I can copy paste and run to understand your intention, use case, and problem.
